internal/proxy/proxy.go

```diff
@@ -68,11 +68,21 @@ const (
 	CacheMetricStore
 )
 
-// Metrics returns a pointer to the current proxy metrics.
-func (p *TransparentProxy) Metrics() *ProxyMetrics {
+// Metrics returns a copy of the current proxy metrics.
+// Returns a value copy to ensure thread-safety when reading metrics.
+func (p *TransparentProxy) Metrics() ProxyMetrics {
```

Comment:
`Metrics()` will panic if `p.metrics` is ever nil (e.g., if someone constructs `TransparentProxy{}` directly or a future refactor changes init order). Consider a defensive guard:

- if `p.metrics == nil`, return `ProxyMetrics{}`

Not strictly required if you guarantee construction via `NewTransparentProxy*`, but it’s a cheap safety net for an exported method.